NYTimesMostViewedArticles/NYTimesMostViewedArticleViewModel.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  NYTimesMostViewedArticleViewModel.swift
+//  NYTimesMostViewedArticles
+//
+//  Created by Sagar Pahwa on 17/11/20.
+//  Copyright © 2020 Sagar Pahwa. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+extension ViewedArticle: ArticleRepresentable {
+    var author: String? {
+        return self.source
+    }
+    
+    var date: String? {
+        return self.publishedDate
+    }
+}
+
+
+class NYTimesMostViewedArticleViewModel {
+    
+    let networkClient: NetworkClientType
+    
+    private(set) var articles = [ViewedArticle]()
+    
+    init(networkClient: NetworkClientType) {
+        self.networkClient = networkClient
+    }
+    
+    func fetchMostViewArticles(completion: @escaping (Error?)->()) {
+        self.networkClient.getMostViewedArticles(period: .day) { (response, error) in
+            if let articleResults = response?.results {
+                DispatchQueue.main.async {
+                    self.articles = articleResults
+                    completion(nil)
+                }
+            }
+            else if let error = error {
+                completion(error)
+            }
+        }
+    }
+    
+    func showDetailScreen(index: Int, source: UIViewControllerType) {
+        if let articleDetailVC = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(identifier: "ArticleDetailVC") as? ArticleDetailView {
```

Comment:
Please declare all strings in a constant.

Reply:
Since this comment hasn't been implemented yet, hence please follow more better solution as explained below,

Since all the UI classes will be subclass of UIResponder hence instead of hardcoding reuseId in each class, we can create an extension of UIResponder and use it.

extension UIResponder { static var identifier: String { return "\(self)" } }

Instead of hardcoding "ArticleDetailVC" we can directly call ArticleDetailViewController.identifier` wherever required.
Please also follow this for ViewedArticleCell.